libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/JsonExtensions.cs

```diff
@@ -18,5 +22,55 @@ public static T Dequeue<T>(this List<T> queue)
 
             return result;
         }
+
+        /// <summary>
+        /// Achieve deep data binding with recursion.
+        /// </summary>
+        /// <param name="token">Input token.</param>
+        /// <param name="state">Memory scope.</param>
+        /// <returns>Deep data binding result.</returns>
+        public static JToken ReplaceJTokenRecursively(this JToken token, object state)
+        {
+            switch (token.Type)
+            {
+                case JTokenType.Object:
+                    // NOTE: ToList() is required because JToken.Replace will break the enumeration.
+                    foreach (var child in token.Children<JProperty>().ToList())
+                    {
+                        child.Replace(child.ReplaceJTokenRecursively(state));
+                    }
+
```

Comment:
Extra line.

Reply:
the extra line should be the one between break and case. I just removed all of them.